### PR TITLE
Adding Application Startup method to be invoked on UI thread

### PR DESCRIPTION
**Pull request by nickrandolph**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Bug fix

### :arrow_heading_down: What is the current behavior?
Initialize is called on background thread which causes issues for any Application init actions that need to be done on UI thread

### :new: What is the new behavior (if this is a feature change)?
Startup method invoked on Application on the UI thread

### :boom: Does this PR introduce a breaking change?
No

### :memo: Links to relevant issues/docs
https://github.com/MvvmCross/MvvmCross/issues/2745

### :thinking: Checklist before submitting

- [ ] All projects build
- [ ] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [ X ] Rebased onto current develop

### CODE REVIEW

File: MvvmCross/ViewModels/MvxAppStart.cs [hunk 1]
```diff
@@ -28,35 +28,60 @@ public void Start(object hint = null)
 
         public virtual void ResetStart()
         {
+            Reset();
             Interlocked.Exchange(ref startHasCommenced, 0);
         }
+
+        protected virtual void Reset()
```
Comment (martijn00) [inline MvvmCross/ViewModels/MvxAppStart.cs]: Should this be empty?

File: MvvmCross/ViewModels/MvxAppStart.cs [hunk 2]
```diff
@@ -28,35 +28,60 @@ public void Start(object hint = null)
...
         }
 
         protected override void Startup(object hint = null)
+        {
+            ApplicationStartup();
+
+            NavigateToFirstViewModel(hint);
+        }
+
+        protected virtual void ApplicationStartup(object hint = null)
```
Comment (martijn00) [inline MvvmCross/ViewModels/MvxAppStart.cs]: Do we need this method? We could just call `Application.Startup(hint);` from `Startup `?

File: MvvmCross/ViewModels/MvxAppStart.cs [hunk 3]
```diff
@@ -28,35 +28,60 @@ public void Start(object hint = null)
...
+        {
+            MvxLog.Instance.Trace("AppStart: Application Startup - On UI thread");
+            Application.Startup(hint);
+        }
+
+        protected virtual void NavigateToFirstViewModel(object hint)
```
Comment (martijn00) [inline MvvmCross/ViewModels/MvxAppStart.cs]: Would it make sense to let this return the `StartupViewModel`, and call that from `Startup`?

**Comment by kjeremy** [comment]
Are we still able to start MvvmCross from a background service?

File: MvvmCross/ViewModels/MvxAppStart.cs (hunk 1 above)
Comment (nickrandolph) [reply MvvmCross/ViewModels/MvxAppStart.cs]: no longer empty in base mvxappstart class

File: MvvmCross/ViewModels/MvxAppStart.cs (hunk 2 above)
Comment (nickrandolph) [reply MvvmCross/ViewModels/MvxAppStart.cs]: I think it makes more sense now after this has been refactored

File: MvvmCross/ViewModels/MvxAppStart.cs (hunk 3 above)
Comment (nickrandolph) [reply MvvmCross/ViewModels/MvxAppStart.cs]: Not sure I understand what you're thinking here - take a look at my refactor and let me know

**Comment by nickrandolph** [comment]
@kjeremy I think that should be fine but if you're able to generate a sample of background service and add to playground, that'd be great.

**Pull request merged**